src/features/Header/Header.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { useNavigate } from "react-router";
+import { useAuth } from "../../context/authContext";
+import { useMutation } from "react-query";
+import { authSignOut } from "../../firebase/auth";
+import { Button, Box } from "@mui/material";
+
+const Header = () => {
+  const navigate = useNavigate();
+  const { isLoggedIn } = useAuth();
+
+  const signOutMutation = useMutation(authSignOut, {
+    onSuccess: () => {
+      navigate("/login");
+    },
+  });
+
+  const handleSignOut = () => {
+    signOutMutation.mutate();
+  };
```

Comment:
Good use of mutations. The only thing could be destructuring the mutate function of useMutation.

```suggestion
  const { mutate: signOut } = useMutation(authSignOut, {
    onSuccess: () => {
      navigate("/login");
    },
  });

  const handleSignOut = () => {
    signOut();
  };
```

Reply:
You won't be needing handleSignOut function anymore because it basically returns the mutate function. You should be able to call `signOut` directly where you called `handleSignOut`.